gh-101100: Fix Sphinx warning in library/http.cookies.rst (#112908)
python/cpython: Doc/library/http.cookies.rst

The module formerly strictly applied the parsing rules described in the
:rfc:`2109` and :rfc:`2068` specifications.  It has since been discovered that
MSIE 3.0x didn't follow the character rules outlined in those specs and also
many current day browsers and servers have relaxed parsing rules when it comes to

> cookie handling.  As a result, the parsing rules used are a bit less strict.

Reviewer: (Minor wordsmithing)

```suggestion
MSIE 3.0x didn't follow the character rules outlined in those specs; many
current-day browsers and servers have also relaxed parsing rules when it comes
to cookie handling.  As a result, this module now uses parsing rules that are a
bit less strict than they once were.
```

.. class:: SimpleCookie([input])

   This class derives from :class:`BaseCookie` and overrides :meth:`~BaseCookie.value_decode`
   and :meth:`~BaseCookie.value_encode`. SimpleCookie supports strings as cookie values.

>    When setting the value, :class:`!SimpleCookie` calls the builtin :func:`str()` to convert

Reviewer: ```suggestion
   and :meth:`~BaseCookie.value_encode`. :class:`!SimpleCookie` supports
   strings as cookie values. When setting the value, :class:`!SimpleCookie`
   calls the builtin :func:`str` to convert
```

   Abstract a key/value pair, which has some :rfc:`2109` attributes.

   Morsels are dictionary-like objects, whose set of keys is constant --- the valid
   :rfc:`2109` attributes, which are:

>    * .. attribute:: expires
>    * .. attribute:: path
>    * .. attribute:: comment
>    * .. attribute:: domain
>    * .. attribute:: max-age
>    * .. attribute:: secure
>    * .. attribute:: version
>    * .. attribute:: httponly
>    * .. attribute:: samesite

Reviewer: Currently this is rendered as a nice compact list by Sphinx:

<details>
<summary>Screenshot</summary>

![image](https://github.com/python/cpython/assets/66076021/94ffcb2d-a12e-46cf-b0bc-23dfe42899f5)

</details>

Unfortunately with this PR, Sphinx adds double spacing between the bullets, which feels like it takes up an annoying amount of vertical space :(

<details>
<summary>Screenshot</summary>

![image](https://github.com/python/cpython/assets/66076021/d3cdd4f6-1335-4d47-9d37-014430807ed8)

</details>

If we did this instead, then we still get the link anchors, but we also get nice compact formatting like we currently have on `main`. We lose the bullet points, but I can't figure out a way of keeping the compact formatting while also keeping the bullets, unfortunately:

```suggestion
     .. attribute:: expires
                    path
                    comment
                    domain
                    max-age
                    secure
                    version
                    httponly
                    samesite
```

How it renders:

<details>
<summary>Screenshot</summary>

![image](https://github.com/python/cpython/assets/66076021/1f5c5a6e-d464-4ecf-9669-6f3709a8f775)

</details>
